### PR TITLE
federationuser(ldap): adding keepalive settings for LDAP connections (PROJQUAY-5137)

### PR DESCRIPTION
**Pull request by michaelalang**
Adding keepalive setting for LDAP connection in complex network scenarios.

introducing variables for 
* _DEFAULT_KEEPALIVE_IDLE
* _DEFAULT_KEEPALIVE_INTERVAL
* _DEFAULT_KEEPALIVE_PROBES

setting the corresponding LDAP options. 
The LDAP restart option ensures on issues that the connection will be re-established.

### CODE REVIEW

File: data/users/externalldap.py
Comment (kleesc) [inline data/users/externalldap.py:22]: Just curious as to how we figured out these were sane defaults. 
What are the values when not explicitly set?

File: data/users/externalldap.py
Comment (michaelalang) [reply data/users/externalldap.py:307]: @kleesc I based the second PR on the first (1974) that's why those lines are in. If you prefer, I can rebase them to the current main.

File: data/users/externalldap.py
Comment (michaelalang) [reply data/users/externalldap.py:22]: the default values are 

KEEPALIVE_IDLE = 7200 # 2 hours
KEEPALIVE_INTERVAL = 75 # seconds
KEEPALIVE_PROBES = 9 # until connection get's closed.

with the `quay-operator` killing the POD within ~2 minutes if the health checks fail, it seems obvious to set them low as considered being internal network connectivity the size of 64b is worth mentioning and recovery should be possible.
The settings reflect a connection after 10 seconds of being IDLE will send very 5 seconds a keepalive and drop/close the connection after worst case 15 seconds no response received. If meanwhile data passes the connection, keepalive will reset for the 10 seconds IDLE.

File: data/users/externalldap.py
Comment (kleesc) [reply data/users/externalldap.py:307]: Yea, I would split these if you think they're not related. Makes it easier to revert if needed. Otherwise LGTM

File: data/users/externalldap.py
Comment (michaelalang) [reply data/users/externalldap.py:22]: do you want me to change/add config.yaml parameters for it ?

Comment (kleesc) [reply data/users/externalldap.py:22]: I don't think that's necessary as you're currently just updating the default values. If you want to make it configurable, you would have to add these to https://github.com/quay/quay/blob/master/data/users/__init__.py#L56-L90.

Comment (michaelalang) [reply data/users/externalldap.py:22]: well I guess we can save that for a different PR  :dancing_women: 
thanks for your input and support, will update the PR accordingly

**Review by kleesc — APPROVED** [review]

**Comment by Sunandadadi** [comment]
@michaelalang Are you waiting on anything to merge this PR? If this is good to go, can you please merge?

**Comment by michaelalang** [comment]
@Sunandadadi I am only contributing not deciding and or merging requests. Please give the guys the time to do so ...

**Pull request merged**
